packages/muelu/test/tutorial/TutorialDriver.cpp

```diff
@@ -0,0 +1,479 @@
+// *****************************************************************************
+// @HEADER
+
+#include <iostream>
+
+#include <Amesos2.hpp>
+#include <Amesos2_config.h>
+
+#include <Xpetra_MultiVectorFactory.hpp>
+#include <Xpetra_ImportFactory.hpp>
+#include <Xpetra_IO.hpp>
+
+#include <BelosConfigDefs.hpp>
+#include <BelosLinearProblem.hpp>
+#include <BelosBlockCGSolMgr.hpp>
+#include <BelosPseudoBlockCGSolMgr.hpp>
+#include <BelosBlockGmresSolMgr.hpp>
+#include <BelosXpetraAdapter.hpp>
+#include <BelosMueLuAdapter.hpp>
+#include <BelosSolverFactory.hpp>
+#include <BelosStatusTestCombo.hpp>
+#include <BelosXpetraStatusTestGenResSubNorm.hpp>
+
+#include <Galeri_XpetraMaps.hpp>
+#include <Galeri_XpetraProblemFactory.hpp>
+#include <Galeri_XpetraUtils.hpp>
+
+#include <MueLu.hpp>
+#include <MueLu_BaseClass.hpp>
+#include <MueLu_ConfigDefs.hpp>
+#include <MueLu_CreateTpetraPreconditioner.hpp>
+#include <MueLu_Hierarchy.hpp>
+#include <MueLu_Level.hpp>
+#include <MueLu_Utilities.hpp>
+
+#include <Teuchos_StandardCatchMacros.hpp>
+#include <Teuchos_XMLParameterListHelpers.hpp>
+
+#include <Tpetra_CrsMatrix.hpp>
+#include <Tpetra_Map.hpp>
+#include <Tpetra_Vector.hpp>
+#include "Teuchos_ArrayViewDecl.hpp"
+
+#include <Xpetra_Map.hpp>
+
+int main(int argc, char* argv[]) {
+  bool success = false;
+  try {
+    Tpetra::ScopeGuard tpetraScope(&argc, &argv);
+    {
+      //! [TpetraTemplateParameters begin]
+      using SC = Tpetra::MultiVector<>::scalar_type;
+      using LO = Tpetra::MultiVector<>::local_ordinal_type;
+      using GO = Tpetra::MultiVector<>::global_ordinal_type;
+      using NO = Tpetra::MultiVector<>::node_type;
+      //! [TpetraTemplateParameters end]
+
+      // ================================
+      // Convenient definitions
+      // ================================
+      //! [UsingStatements begin]
+      using Teuchos::ParameterList;
+      using Teuchos::RCP;
+      using Teuchos::rcp;
+      using Teuchos::TimeMonitor;
+
+      using CrsMatrix   = Tpetra::CrsMatrix<>;
+      using Map         = Tpetra::Map<>;
+      using MultiVector = Tpetra::MultiVector<>;
+      using Operator    = Tpetra::Operator<>;
+
+      using STS            = Teuchos::ScalarTraits<SC>;
+      using magnitude_type = typename Teuchos::ScalarTraits<SC>::magnitudeType;
+
+      using real_type             = typename STS::coordinateType;
+      using RealValuedMultiVector = Tpetra::MultiVector<real_type, LO, GO, NO>;
+
+      const SC one = Teuchos::ScalarTraits<SC>::one();
+      //! [UsingStatements end]
+
+      //! [CommunicatorObject begin]
+      RCP<const Teuchos::Comm<int>> comm = Teuchos::DefaultComm<int>::getComm();
+      int MyPID                          = comm->getRank();
+      int NumProc                        = comm->getSize();
+      (void)MyPID;    // unused void pointer cast to avoid unused variable warnings
+      (void)NumProc;  // unused void pointer cast to avoid unused variable warnings
```

Comment:
Would this also work?
```suggestion
      [[maybe_unused]] MyPID;
      [[maybe_unused]] NumProc;
```

Reply:
```
warning: statement has no effect [-Wunused-value]
   91 |       [[maybe_unused]] MyPID;
```
I don't think so.